### DISCUSSION TITLE
Support for supabase-js v2?

### DISCUSSION THREAD

**Discussion by chimame** [mitchelvanbever/remix-auth-supabase#43] [Ideas]
Thanks for the great library.

supabase-js v2 has been released.
https://github.com/supabase/supabase-js/releases/tag/v2.0.0

`remix-auth-supabase` has many things defined in the v1 return value of supabase-js. It is difficult to use `remix-auth-supabase` and v2 of supabase-js at the same time.
A new version development using v2 of supabase-js with `remix-auth-supabase` is needed.
https://supabase.com/docs/reference/javascript/upgrade-guide

I can help with updating to v2 of supabase-js. Will `remix-auth-supabase` update supabase-js to v2 and continue development?

**Comment by JoaoCnh** [comment]
I was able to successfully use the strategy by simply patching the package. By simple replacing `auth.api.getUser` by `auth.getUser`

**Comment by rphlmr** [comment]
Supabase has released its own Remix plugin : https://supabase.com/docs/guides/auth/auth-helpers/remix

I don't know what @mitchelvanbever wants for the future of this lib.

**Reply by mitchelvanbever** [reply]
I've been out of the loop in regards to remix and supabase, so I wasn't aware they now haven an "official" helper package. 

Seems like that should be the recommended way forward. Will take some time to update everything to V2 before archiving the repo and point everyone towards the official helper package.

**Comment by quangv** [comment]
Fwiw, Remix is now at v2 as well, I am getting this error atm:

```
ERR! Found: @remix-run/server-runtime@2.0.1
...
npm ERR! Conflicting peer dependency: @remix-run/server-runtime@1.7.5
npm ERR! node_modules/@remix-run/server-runtime
npm ERR!   peer @remix-run/server-runtime@"1.7.5" from remix-auth-supabase@4.0.2
npm ERR!   node_modules/remix-auth-supabase
```

Thanks for this discussion, I'll look at Supabase's remix plugin
